src/api.js

```diff
@@ -251,12 +251,22 @@ var api = {
     var requestFunction = function() {
       return _request(reqOptions, options.logOptions);
     };
+
+    var secureRequest = true;
+    if (options.origin && options.origin.startsWith("http://")) {
```

Comment:
Maybe use the URL module? https://nodejs.org/docs/latest-v8.x/api/url.html#url_constructor_new_url_input_base

You can use `.protocol` to check for `http:`

Reply:
Good idea.